plugins/modules/zos_gather_facts.py

```diff
@@ -0,0 +1,280 @@
+        facts that do not match 'parmlib'
+  ibm.ibm_zos_core.zos_gather_facts:
+    gather_subset:
+      - ipl
+      - sys
+    filter:
+      - "*parmlib*"
+'''
+
+RETURN = r'''
+ansible_facts:
+  description: Store the facts that are gathered from z/OS systems.
+  returned: sometimes
+  type: dict
+'''
+
+from fnmatch import fnmatch
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.ibm.ibm_zos_core.plugins.module_utils import (
+    zoau_version_checker
+)
+
+
+def zinfo_cmd_string_builder(gather_subset):
+    """Builds a command string for 'zinfo' based off the gather_subset list.
+    Arguments:
+        gather_subset {list} -- A list of subsets to pass in.
+    Returns:
+        [str] -- A string that contains a command line argument for calling
+                 zinfo with the appropriate options.
+        [None] -- An invalid value was received for the subsets.
+    """
+    if gather_subset is None or 'all' in gather_subset:
+        return "zinfo -j -a"
+
+    # base value
+    zinfo_arg_string = "zinfo -j"
+
+    # build full string
+    for subset in gather_subset:
+        # remove leading/trailing spaces
+        subset = subset.strip()
+        # empty string as subset is bad
+        if not subset:
+            return None
+        # sanitize subset against malicious (probably alphanumeric only?)
+        if not subset.isalnum():
+            return None
+        zinfo_arg_string += " -t " + subset
+
+    return zinfo_arg_string
+
+
+def flatten_zinfo_json(zinfo_dict):
+    """Removes one layer of mapping in the dictionary. Top-level keys correspond
+        to zinfo subsets and are removed.
+    Arguments:
+        zinfo_dict {dict} -- A dictionary that contains the parsed result from
+                             the zinfo json string.
+    Returns:
+        [dict] -- A flattened dictionary.
+    """
+    d = dict()
+    for subset in list(zinfo_dict):
+        d.update(zinfo_dict[subset])
+    return d
+
+
+def apply_filter(zinfo_dict, filter_list):
+    """Returns a dictionary that contains only the keys which fit the specified
+       filters.
+    Arguments:
+        zinfo_dict {dict} -- A flattened dictionary that contains results from
+                             zinfo.
+        filter_list {list} -- A string list of shell wildcard patterns (i.e.
+                              'filters') to apply to the zinfo_dict keys.
+    Returns:
+        [dict] -- A dictionary with keys that are filtered out.
+    """
+
+    if filter_list is None or filter_list == [] or '*' in filter_list:
+        return zinfo_dict
+
+    filtered_d = {}
+    for filter in filter_list:
+        for k, v in zinfo_dict.items():
+            if(fnmatch(k, filter)):
+                filtered_d.update({k: v})
+    return filtered_d
+
+
+def run_module():
+    # define available arguments/parameters a user can pass to the module
+    module_args = dict(
+        gather_subset=dict(
```

Comment:
Given it more thought, updates to Automation Hub no longer require us to support Ansible 2.9,  more specifically we only need to support Ansible 2.11.  Since this is planned to release near the end of the year, we might as well just bump the meta to require 2.11 and not invest any more time in validating if this works with Ansible versions less than 2.11, users will get a warning when they try to.